controllers/cluster_create.go

```diff
@@ -1137,6 +1139,10 @@ func (r *ClusterReconciler) reconcilePVCs(
 	}
 
 	pvc := resources.getPVC(pvcToReattach)
+	if pvc == nil {
+		return ctrl.Result{}, fmt.Errorf(
+			"a pvc to reattach was nominated but it could not be found from the pvc list")
```

Comment:
Let's improve this error message too, including the name of the PVC we didn't found.